code/modules/reagents/chemistry/recipes.dm

```diff
@@ -132,3 +282,107 @@
 			else
 				if(step_towards(X, T) && moving_power > 1)
 					addtimer(CALLBACK(GLOBAL_PROC, .proc/_step_towards, X, T), 2)
+
+//////////////////Generic explosions/failures////////////////////
+// It is HIGHLY, HIGHLY recomended that you consume all/a good volume of the reagents/products in an explosion - because it will just keep going forever until the reaction stops
+//If you have competitive reactions - it's a good idea to consume ALL reagents in a beaker (or product+reactant), otherwise it'll swing back with the deficit and blow up again
+
+
+//Spews out the inverse of the chems in the beaker of the products/reactants only
+/datum/chemical_reaction/proc/explode_invert_smoke(datum/reagents/holder, datum/equilibrium/equilibrium, clear_products = TRUE, clear_reactants = TRUE)
+	var/datum/reagents/invert_reagents = new (2100, NO_REACT)//I think the biggest size we can get is 2100?
+	var/datum/effect_system/smoke_spread/chem/smoke = new()
+	var/sum_volume = 0
+	invert_reagents.my_atom = holder.my_atom //Give the gas a fingerprint
+	for(var/datum/reagent/reagent in holder.reagent_list) //make gas for reagents, has to be done this way, otherwise it never stops Exploding
```

Comment:
```suggestion
	for(var/datum/reagent/reagent as anything in holder.reagent_list) //make gas for reagents, has to be done this way, otherwise it never stops Exploding
```